Race conditions in `test_lifetime.py`

### Description

The tests in `test_lifetime.py` all modify and use the config, which can lead to flaky behavior and failed tests depending on the hardware they are run on. 

Example:
https://github.com/erlingstaff/rucio/actions/runs/9936246547/job/27444064430
https://github.com/erlingstaff/rucio/actions/runs/9936246547/job/27450259662

### Steps to reproduce

Depending on your machine, these tests may or may not fail. They fail for me locally (docker compose dev stack, M1 Max 2021), and they did once in the CI, as mentioned above.

To fail the tests, run the following command in a local development setup. They may also fail in the CI:
`tools/pytest.sh -v --tb=short /opt/rucio/tests/test_lifetime.py`

To fix:
Add the `noparallel` decorator to the tests in question

The race conditions can be seen by adding a `time.sleep()` block in the tests. Different sleep times lead to reproducible test fails:

time.sleep(0.5) in the beginning of `test_lifetime_creation_core` reproducibly leads to:

`FAILED tests/test_lifetime.py::test_lifetime_truncate_expiration - rucio.common.exception.UnsupportedOperation: The resource doesn't support the requested operation.
`

time.sleep(0.05) in the beginning of `test_lifetime_creation_core` reproducibly leads to:

`FAILED tests/test_lifetime.py::test_lifetime_creation_client - Failed: DID NOT RAISE <class 'rucio.common.exception.UnsupportedOperation'>`

The tests pass when using time.sleep to add a >1s delay between them

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/tests/test_lifetime.py
+++ b/tests/test_lifetime.py
@@ -31,6 +31,7 @@ from rucio.db.sqla.constants import DIDType
 from rucio.tests.common import skip_multivo
 
 
+@pytest.mark.noparallel(reason='Race conditions with the other tests in test_lifetime.py, they all modify and use the config.')
 @skip_multivo(reason='only valid for ATLAS')
 def test_lifetime_creation_core(root_account, rse_factory, mock_scope, did_factory):
     """
@@ -98,6 +99,7 @@ def test_lifetime_creation_core(root_account, rse_factory, mock_scope, did_facto
         assert did in list_exceptions
 
 
+@pytest.mark.noparallel(reason='Race conditions with the other tests in test_lifetime.py, they all modify and use the config.')
 @skip_multivo(reason='only valid for ATLAS')
 def test_lifetime_truncate_expiration(root_account, rse_factory, mock_scope, did_factory, rucio_client):
     """
@@ -130,6 +132,7 @@ def test_lifetime_truncate_expiration(root_account, rse_factory, mock_scope, did
     assert exception['expires_at'] - exception['created_at'] > timedelta(29)
 
 
+@pytest.mark.noparallel(reason='Race conditions with the other tests in test_lifetime.py, they all modify and use the config.')
 @skip_multivo(reason='only valid for ATLAS')
 def test_lifetime_creation_client(root_account, rse_factory, mock_scope, did_factory, rucio_client):
     """

